plugins/catalog-backend/src/ingestion/processors/BitbucketDiscoveryProcessor.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Logger } from 'winston';
+import { Config } from '@backstage/config';
+
+import { ScmIntegrations } from '@backstage/integration';
+import { LocationSpec } from '@backstage/catalog-model';
+import { BitbucketClient, pageIterator } from './bitbucket';
+import { CatalogProcessor, CatalogProcessorEmit } from './types';
+import { results } from './index';
+
+export class BitbucketDiscoveryProcessor implements CatalogProcessor {
+  private readonly integrations: ScmIntegrations;
+  private readonly logger: Logger;
+
+  static fromConfig(config: Config, options: { logger: Logger }) {
+    const integrations = ScmIntegrations.fromConfig(config);
+
+    return new BitbucketDiscoveryProcessor({
+      ...options,
+      integrations,
+    });
+  }
+
+  constructor(options: { integrations: ScmIntegrations; logger: Logger }) {
```

Comment:
```suggestion
  constructor(options: { integrations: ScmIntegrationRegistry; logger: Logger }) {
```

might as well use the interface here, easier for mocking etc. Same for the private field above

Reply:
Thanks. I have updated the code to use the interface instead